### ISSUE TITLE
Multithreading

### ISSUE THREAD

**Issue by btzr-io** [lbryio/lbry-desktop#1827]
## The Issue
There are some performance issues while rendering / parsing non-playable files

### Expected Behaviour
:rocket: Non-blocking UI events.

### Suggested Solutions
https://electronjs.org/docs/tutorial/multithreading

**Comment by btzr-io** [comment]
See: https://github.com/lbryio/lbry-desktop/pull/1558#issuecomment-406178533

**Comment by btzr-io** [comment]
https://github.com/mrdoob/three.js/issues/13664

**Comment by jessopb** [comment]
I wanted to add some intel to this: 
I noticed the UI blocks intermittently: random delays lock up the entire UI interrupting tooltips, scrolling, input field entry, etc. I'm told that this is well-known, and that it's not unique to LBRY.
I recorded behavior using javascript profiler in tools. 
It looks like _compress from iz-string.js is being called in a blocking way and isn't unblocked until createPersistor.js finishes.

![compresslag](https://user-images.githubusercontent.com/36554050/45567870-fa291880-b828-11e8-9732-b442de461c73.png)
![jsprofiler](https://user-images.githubusercontent.com/36554050/45567871-fa291880-b828-11e8-8e86-a463c1a43b3e.png)

**Comment by tzarebczan** [comment]
@jessopb I'm experiencing the same. @seanyesmunt this is what I was referring to in my DMs. Something definitely introduced in 0.25 as it doesn't happen in 0.24. I'll see if i can pin this to the electron upgrade or something else we changed.

**Comment by tzarebczan** [comment]
@seanyesmunt and I did a little more digging and it's related to how much localstorage an install is using and claim caching. It's happening more often now that we are resolving/caching ~20 claims on each file page with recommended content. I'll create a separate issue for this to try to unblock the call or improve caching logic/performance. Clearing the cache (settings option) improves this temporarily until more claims are cached. 

Easy way to reproduce fill up the cache - change search results to 1000 and try a new search query (won't automatically update current query - this is logged as an issue).

**Comment by jessopb** [comment]
Is it worth removing 'claims' from the whitelist in store.js? Or does that defeat the main purpose of persist?

**Comment by seanyesmunt** [comment]
I think it's worth trying to just remove that. The resolve calls are way faster now so it should be ok.
